### DISCUSSION TITLE
How to handle env vars in Azure web apps?

### DISCUSSION THREAD

**Discussion by jmallberg** [vercel/next.js#19661] [Help]
I am currently having some problems using environmental variables that is set through app setting. 
The variables are available to the server and we can console.log them out, but we can't get them to show up client side.

Does anyone have some experience dealing with this?

Just to give you some examples of what we have tried;
- Adding NEXT_PUBLIC_ prefixes to the variables
- Exposing them through the next.config.js
- Exposing them in publicRuntimeConfig

All of this works locally, but not when running in Azure!

**Comment by jamesmosier** [comment]
Are you building your Next app in Azure with the environment variables present at the time of build?

**Reply by jmallberg** [reply]
We are trying to set an apolloClient endpoint that points to different CMS environments. Our problem now is that the env variables that are passed in on runtime by Azure are not available to our application Client side.

Is there any way to bypass this problem?

**Reply by jamesmosier** [reply]
`publicRuntimeConfig` would be the only way, bu the env vars must be present before you run `next start`. Changing them on the fly while the server is running will not work.
